src/Microsoft.SqlTools.ServiceLayer/Program.cs

```diff
@@ -39,6 +39,12 @@ internal static void Main(string[] args)
                 }
 
                 Logger.Initialize(tracingLevel: commandOptions.TracingLevel, logFilePath: logFilePath, traceSource: "sqltools", commandOptions.AutoFlushLog);
+                // Only enable SQL Client logging when verbose or higher to avoid extra overhead when the
+                // detailed logging it provides isn't needed
+                if (Logger.TracingLevel.HasFlag(SourceLevels.Verbose))
+                {
+                    new SqlClientListener();
```

Comment:
how do you guarantee this doesn't get garbage collected before the MDS eventsource is created? Is it because `EventListener` constructor registers itself for eventsource notifications right away?

Reply:
That should be enough, but you reminded me I meant to do this anyways....